### PR TITLE
LP1740196: Add tracks from browse view to crates and playlists

### PR DESCRIPTION
**Pull request by uklotzde**
https://bugs.launchpad.net/mixxx/+bug/1740196
https://bugs.launchpad.net/mixxx/+bug/1740619

### CODE REVIEW

File: src/library/browse/browsetablemodel.cpp [hunk 1]
```diff
@@ -130,9 +130,15 @@ QString BrowseTableModel::getTrackLocation(const QModelIndex& index) const {
...
 
 TrackId BrowseTableModel::getTrackId(const QModelIndex& index) const {
-    Q_UNUSED(index);
-    // We can't implement this as it stands.
-    return TrackId();
+    TrackPointer pTrack = getTrack(index);
+    if (pTrack) {
+        return pTrack->getId();
+    } else {
+        qWarning()
```
Comment (daschuer) [inline src/library/browse/browsetablemodel.cpp]: I think the track should be added to the library in this case. We do this for Banshee for example here: 
https://github.com/mixxxdj/mixxx/blob/master/src/library/banshee/bansheeplaylistmodel.cpp#L289

Comment (uklotzde) [reply src/library/browse/browsetablemodel.cpp]: getTrack() either gets the track from the library by location or adds it. The warning only appears if neither of both is possible.

Comment (daschuer) [reply src/library/browse/browsetablemodel.cpp]: Ups, thank you for clarifying.

**Comment by daschuer** [comment]
LGTM, thank you.

**Pull request merged**